src/Impl/Utils/MultipleJsonsSplitter.php

```diff
@@ -39,10 +40,16 @@ static function splitToSingleJsons(string $jsonsString): array {
                     $currentJsonLength = $i - $currentJsonStartIndex + 1;
                     $resultJson = substr($escapedJsonsString, $currentJsonStartIndex, $currentJsonLength);
                     $resultJsons[] = self::unmaskEscapedQuotes($resultJson);
+                    $endIndexOfLastValidSingleJson = $i;
                 }
             }
         }
 
+        $remainingString = trim(substr($escapedJsonsString, 1 + $endIndexOfLastValidSingleJson));
+        if (!empty($remainingString)) {
+            $resultJsons[] = self::unmaskEscapedQuotes($remainingString);
+        }
```

Comment:
Added this for the case when the response is suddenly broken (so it is not a valid json, and was not collected in the main loop above).

Still return that one in the array of separated jsons

Reply:
case of unexpected response from api should cause the exception in publisher.send

Reply:
Yes, we have such test, see this comment: https://github.com/streamx-dev/streamx-ingestion-php/pull/6#discussion_r1881892536